packages/compass-aggregations/src/modules/update-view.ts

```diff
@@ -107,6 +112,25 @@ export const updateView = (): PipelineBuilderThunkAction<Promise<void>> => {
       getState(),
       pipelineBuilder
     );
+
+    // confirmation banner if indexes present
+    await dispatch(fetchIndexes(viewNamespace));
```

Comment:
This dispatch will immediately resolve if you're still in the process of fetching the indexes, so the way it is currently implemented you can't rely on this fetch to ensure that indexes are fetched and stored in the state